fe/fe-core/src/main/java/com/starrocks/sql/common/ListPartitionDiffer.java

```diff
@@ -14,18 +14,46 @@
 
 package com.starrocks.sql.common;
 
+import com.google.common.collect.ImmutableSortedMap;
+import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import com.starrocks.catalog.Column;
+import com.starrocks.catalog.MaterializedView;
+import com.starrocks.catalog.Table;
+import com.starrocks.common.util.DebugUtil;
+import com.starrocks.connector.PartitionUtil;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.List;
 import java.util.Map;
+import java.util.Set;
+
+import static com.starrocks.sql.optimizer.OptimizerTraceUtil.logMVPrepare;
 
 public final class ListPartitionDiffer extends PartitionDiffer {
     private static final Logger LOG = LogManager.getLogger(ListPartitionDiffer.class);
 
     /**
      * Iterate srcListMap, if the partition name is not in dstListMap or the partition value is different, add into result.
      *
+     * Compare the partition of the base table and the partition of the mv.
+     * @param baseItems the partition name to its list partition cell of the base table
+     * @param mvItems the partition name to its list partition cell of the mv
+     * @return the list partition diff between the base table and the mv
+     */
+    public static ListPartitionDiff getListPartitionDiff(Map<String, PListCell> baseItems,
+                                                         Map<String, PListCell> mvItems) {
+        // This synchronization method has a one-to-one correspondence
+        // between the base table and the partition of the mv.
+        Map<String, PListCell> adds = diffList(baseItems, mvItems);
```

Comment:
any risk to be a time-consuming step when table or mv has a large number of partitions?

Reply:
Yep.  It's possible to be time consuming.

I will add a benchmark to figure out how slow it's, and do some optimization later in the next pr.